src/GoTrueClient.ts

```diff
@@ -530,28 +530,57 @@ export default class GoTrueClient {
   }
 
   /**
-   * Sets the session data from refresh token and returns current session or an error if the refresh token is invalid.
-   * @param refresh_token A refresh token returned by supabase auth.
+   * Sets the session data from the current session. If the current session is expired, setSession will take care of refreshing it to obtain a new session.
+   * If the refresh token in the current session is invalid and the current session has expired, an error will be thrown.
+   * If the current session does not contain at expires_at field, setSession will use the exp claim defined in the access token.
+   * @param currentSession The current session that minimally contains an access token, refresh token and a user.
    */
-  async setSession(refresh_token: string): Promise<AuthResponse> {
+  async setSession(refresh_token: string, access_token = ''): Promise<AuthResponse> {
```

Comment:
@thorwebdev i decided to default the `access_token` here to an empty string so this wouldn't be a breaking change

Reply:
@kangmingtay I'd prefer making the breaking change now and changing the argument to an object, which makes it easier to make changes in the future. Wdyt?

Reply:
Yes I agree with @thorwebdev. An object argument is easier to expand on later in a backward compatible way...

Reply:
yup sounds good, will make the change!